go/oasis-node/cmd/config/migrate/migrate.go

```diff
@@ -724,13 +725,41 @@ func doMigrateConfig(cmd *cobra.Command, args []string) {
 
 	// If a node has `consensus.validator` set to true and it does not have any runtimes configured,
 	// the new `mode` should be set to `validator`.
+	var isValidator bool
 	if newValidator, ok := m(newCfg["consensus"])["validator"]; ok {
-		isValidator, _ := newValidator.(bool)
+		isValidator, _ = newValidator.(bool)
 		if _, hasRuntimes := m(newCfg["runtime"])["paths"]; !hasRuntimes && isValidator {
 			nodeMode = "validator"
 			delete(m(newCfg["consensus"]), "validator")
 		}
 	}
+	// If node is a validator, it now requires external P2P addresses to have set.
+	if isValidator {
+		// Set P2P registration addresses based on consensus.external_address.
+		ea := m(newCfg["consensus"])["external_address"]
+		if eaStr, ok := ea.(string); !ok {
+			logger.Warn("consensus.external_address missing, not configuring p2p registration parameters", "address", ea)
+		} else {
+			// Parse host from the external address.
+			url, err := url.Parse(eaStr)
+			if err != nil {
+				logger.Error("failed to parse URI from consensus.external_address", "err", err)
+				os.Exit(1)
+			}
+			// Set P2P port if not set.
+			if _, ok := m(newCfg["p2p"])["port"]; !ok {
+				m(newCfg["p2p"])["port"] = 9200
+			}
+			// Set P2P registration addresses if not set.
```

Comment:
Would it make sense for this to append the address in the (weird) edge case where "p2p.port" would not be set, but "p2p.registration.addresses" would be set?

Reply:
Yeah maybe, but then we'd need to check again that the port used there is not already used for CometBFT.

Reply:
Yeah, maybe just do nothing in that case and just emit a warning?

Reply:
Done :+1: